vms/platformvm/warp/benchmarks/environment_test.go

```diff
@@ -0,0 +1,413 @@
+	"github.com/ava-labs/avalanchego/utils/json"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/version"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/api"
+	"github.com/ava-labs/avalanchego/vms/platformvm/config"
+	"github.com/ava-labs/avalanchego/vms/platformvm/fx"
+	"github.com/ava-labs/avalanchego/vms/platformvm/metrics"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs/executor"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs/mempool"
+	"github.com/ava-labs/avalanchego/vms/platformvm/utxo"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+
+	blockexecutor "github.com/ava-labs/avalanchego/vms/platformvm/blocks/executor"
+	p_tx_builder "github.com/ava-labs/avalanchego/vms/platformvm/txs/builder"
+	pvalidators "github.com/ava-labs/avalanchego/vms/platformvm/validators"
+)
+
+const (
+	testNetworkID = 10 // To be used in tests
+	defaultWeight = 10000
+)
+
+var (
+	_ mempool.BlockTimer = (*environment)(nil)
+
+	defaultMinStakingDuration = 24 * time.Hour
+	defaultMaxStakingDuration = 365 * 24 * time.Hour
+	defaultGenesisTime        = time.Date(1997, 1, 1, 0, 0, 0, 0, time.UTC)
+	defaultValidateStartTime  = defaultGenesisTime
+	defaultValidateEndTime    = defaultValidateStartTime.Add(10 * defaultMinStakingDuration)
+	defaultMinValidatorStake  = 5 * units.MilliAvax
+	defaultBalance            = 100 * defaultMinValidatorStake
+	preFundedKeys             = secp256k1.TestKeys()
+	avaxAssetID               = ids.ID{'y', 'e', 'e', 't'}
+	defaultTxFee              = uint64(100)
+	xChainID                  = ids.Empty.Prefix(0)
+	cChainID                  = ids.Empty.Prefix(1)
+
+	genesisBlkID ids.ID
+
+	errMissingPrimaryValidators = errors.New("missing primary validator set")
+	errMissing                  = errors.New("missing")
+)
+
+type environment struct {
+	blkManager blockexecutor.Manager
+	mempool    mempool.Mempool
+	sender     *common.SenderTest
+
+	isBootstrapped    *utils.Atomic[bool]
+	config            *config.Config
+	clk               *mockable.Clock
+	baseDB            *versiondb.Database
+	ctx               *snow.Context
+	fx                fx.Fx
+	state             state.State
+	atomicUTXOs       avax.AtomicUTXOManager
+	uptimes           uptime.Manager
+	utxosHandler      utxo.Handler
+	txBuilder         p_tx_builder.Builder
+	backend           *executor.Backend
+	validatorsManager pvalidators.Manager
+}
+
+func (*environment) ResetBlockTimer() {
+	// dummy call, do nothing for now
+}
+
+func newEnvironment(tb testing.TB, subnetIDToRegister ids.ID) *environment {
```

Comment:
Why do we need to set this up instead of just using `defaultVM()` ?

It seems like the following would work just as well and remove a lot of this code:

1. Rename `defaultVM()` to `defaultVMWithDBManager(manager manager.Manager)`
2. Create a new `defaultVM()` that passes in a memory database manager
3. For any new test that needs to operate on leveldb, use `defaultVMWithDBManager(leveldbManagerInstance)`

Reply:
If `defaultVM()` is currently insufficient, I'd rather update it to support the use case we need than copy pasting similar setup code